File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1

  def self.compute(strand_1, strand_2)
    raise ArgumentError, ("Strands must be the same length") if strand_1.length != strand_2.length

    return 0 if strand_1 == strand_2

    0.upto(strand_1.length).count do |index|
      strand_1[index] != strand_2[index]
    end
  end
end

File: javascript/hamming/hamming.js
var Hamming = function(){};

Hamming.prototype.compute = function(string1, string2){
  if (string1.length !== string2.length) {
    throw "DNA strands must be of equal length."
  };
  
  count = 0;
  for (var i = 0; i < string1.length; i++) {
    if (string1[i] !== string2[i]) {
      count += 1;
    }
  }
  return count;
};

module.exports = Hamming;
